utils/error_handling_helper.go

```diff
@@ -3,22 +3,16 @@ package utils
 import (
 	"errors"
 
-	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/smithy-go"
 )
 
 // Helper method to check if the request is fatal/4XX status
 func IsFatalError(errMsg error) bool {
 
-	if reqErr, ok := errMsg.(awserr.RequestFailure); ok {
+	var ae smithy.APIError
+	if errors.As(errMsg, &ae) {
 		// check if client side error occurred
-		if reqErr.StatusCode() >= 400 && reqErr.StatusCode() < 500 {
-			return true
-		}
-	}
-	if reqErr, ok := errMsg.(awserr.Error); ok {
-		if reqErr.OrigErr() != nil {
-			return IsFatalError(reqErr.OrigErr())
-		}
+		return ae.ErrorFault() == smithy.FaultClient
```

Comment:
Do you have docs that show this is equivalent to all HTTP 400 errors? Do 403s show up as a smithy.FaultClient for example?

Reply:
> smithy.FaultClient represents all client-side errors (HTTP 4XX responses, including 403s).

I might've missed it, I'm not seeing this in the docs or the code linked.